src/main/java/com/opentable/db/postgres/embedded/EmbeddedPostgres.java

```diff
@@ -433,15 +442,91 @@ private void mkdirs(File dir)
     private static final AtomicReference<File> BINARY_DIR = new AtomicReference<>();
     private static final Lock PREPARE_BINARIES_LOCK = new ReentrantLock();
 
-    private File prepareBinaries(PgBinaryResolver pgBinaryResolver) {
+    /**
+     * Get current operating system string. The string is used in the appropriate postgres binary name.
+     *
+     * @return Current operating system string.
+     */
+    private static String getOS(){
+        return SystemUtils.IS_OS_WINDOWS ? "Windows" : system("uname", "-s").get(0);
+    }
+
+    /**
+     * Get the machine architecture string. The string is used in the appropriate postgres binary name.
+     *
+     * @return Current machine architecture string.
+     */
+    private static String getArchitecture(){
+        if (!SystemUtils.IS_OS_WINDOWS) {
+            return system("uname", "-m").get(0);
+        }
+
+        return "amd64".equals(SystemUtils.OS_ARCH) || SystemUtils.OS_ARCH == null ? "x86_64" : SystemUtils.OS_ARCH;
+    }
+
+    /**
+     * Unpack archive compressed by tar with bzip2 compression. By default system tar is used (faster). If not found, then the
+     * java implementation takes place.
+     *
+     * @param tbzPath The archive path.
+     * @param targetDir The directory to extract the content to.
+     */
+    private static void extractTbz(final String tbzPath, final String targetDir) throws IOException {
+        try {
+            system("tar", "-x", "-f", tbzPath, "-C", targetDir);
+        } catch (final Exception e) {
+            try (
```

Comment:
What if this exception is because of a corrupt download, or truncated file, or something?  It'd get swallowed.  Probably should just get rid of the fallback and use commons-compress directly.